synapse/axon.py

```diff
@@ -1,801 +1,1153 @@
-import time
-import lmdb
+import enum
+import lmdb  # type: ignore
+import asyncio
+import binascii
 import struct
 import logging
+import random
 import hashlib
-import itertools
+import tempfile
+import threading
+import concurrent
+import contextlib
+import collections
 
+import synapse.exc as s_exc
 import synapse.glob as s_glob
 import synapse.common as s_common
+import synapse.daemon as s_daemon
 import synapse.eventbus as s_eventbus
+import synapse.telepath as s_telepath
 
 import synapse.lib.cell as s_cell
+import synapse.lib.coro as s_coro
 import synapse.lib.lmdb as s_lmdb
 import synapse.lib.const as s_const
 
 logger = logging.getLogger(__name__)
 
-zero64 = b'\x00\x00\x00\x00\x00\x00\x00\x00'
-blocksize = s_const.mebibyte * 64
+# File convention: bsid -> unique id for a blobstor (actually the cell iden)
 
-class BlobStor(s_eventbus.EventBus):
+
+CHUNK_SIZE = 16 * s_const.mebibyte
+
+async def to_aiter(it):
+    '''
+    Take either a sync or async iteratable and yields as an async generator
     '''
-    The blob store maps buid,indx values to sequences of bytes stored in a LMDB database.
+    if hasattr(it, '__aiter__'):
+        async for i in it:
+            yield i
+    else:
+        for i in it:
+            yield i
+
+def _find_hash(curs, key):
     '''
+    Set a LMDB cursor to the first key that starts with \a key
```

Comment:
what is a \a key?

Reply:
in doxygen, it highlights it to mean the parameter variable.